packages/conda-store/src/condaStore.ts

```diff
@@ -266,14 +275,56 @@ export async function createEnvironment(
   environment: string,
   dependencies: Array<string>
 ): Promise<void> {
-  await specifyEnvironment(
+  const specification: ICondaStoreSpecification = {
+    name: environment,
+    dependencies
+  };
+  await specifyEnvironment(baseUrl, namespace, stringify(specification));
+  return;
+}
+
+/**
+ * Clone an existing environment.
+ *
+ * @async
+ * @param {string} baseUrl - Base URL of the conda-store server; usually http://localhost:5000
+ * @param {string} existingNamespace - Namespace of the existing environment.
+ * @param {string} existingEnvironment - Name of the existing environment.
+ * @param {string} namespace - Namespace into which the clone will be created.
+ * @param {string} environment - Name of the new environment, which will be a clone of the existing environment.
+ * @param {Array<string>} dependencies - List of string package names to be added to the spec.
+ * @returns {Promise<void>}
+ */
+export async function cloneEnvironment(
+  baseUrl: string,
+  existingNamespace: string,
+  existingEnvironment: string,
+  namespace: string,
+  environment: string
+): Promise<void> {
+  // Get specification for existing environment
+  const specificationResponse = await exportEnvironment(
+    baseUrl,
+    existingNamespace,
+    existingEnvironment
+  );
+
+  // Modify specification so that it uses the provided environment name
+  const specificationYaml = await specificationResponse.text();
+  const specification: ICondaStoreSpecification = parse(specificationYaml);
+  specification.name = environment;
+
+  // Pass specification to the API to create new environment based on the
+  // provided existiing environment
+  const response = await specifyEnvironment(
     baseUrl,
     namespace,
-    stringify({
-      name: environment,
-      dependencies
-    })
+    stringify(specification)
```

Comment:
There is another way this could be done:

```ts
const response = await createEnvironment(
  baseUrl,
  namespace,
  environment,
  specification.dependencies
);
```

Not sure which way is better. With the current method, everything from the build-specification-yaml file will be copied (three fields: channels, dependencies, prefix). With the alternative method above, just one field (dependencies) will be copied.